packages/vitest/src/node/pools/typecheck.ts

```diff
@@ -92,7 +92,7 @@ export function createTypecheckPool(ctx: Vitest): ProcessPool {
 
   async function startTypechecker(project: WorkspaceProject, files: string[]) {
     if (project.typechecker) {
-      return project.typechecker
+      return project.typechecker.prepare()
```

Comment:
I am confused why this is needed 🤔 

If there is a typechecked, then `prepare` was called already 🤔

Reply:
It's probably not strictly necessary right now, but I put it there for a couple of reasons:

1. `project.typechecker = ...` is called early in the `createWorkspaceTypechecker` function, so depending on how this is called (which might be optimised/changed in future) it *isn't* necessarily true that `.prepare()` has been called, and resolved, by the time the typechecker prop is set.
2. `promises` is an array of `Promise<void>`, so returning `project.typechecker` here was causing a type error when pushing this into the array. prepare() returns void so is ok to return here.

I had a look at the implementation of prepare() and the only async op in it is reading the tsconfig, no scary side effects. So it thought the safest thing would be to just call it again to be 100% sure it's fully done.

So, options are:

a) Just `return` here
b) Change the the `createWorkspaceTypechecker` to set the typechecker property here
c) Leave it like this

I'm fine with any of the above.

Reply:
I would just keep a single `return` here

Reply:
Done